packages/mask/src/extension/popups/hook/useFriends.ts

```diff
@@ -43,90 +44,94 @@ export const PlatformSort: Record<NextIDPlatform, number> = {
-            {
-                network: 'all',
-                pageOffset: 0,
-            },
-            1000,
-        )
-        if (values.length === 0) return EMPTY_LIST
-        const friends: Friend[] = (
-            await Promise.all<Promise<Friend | undefined>>(
-                values.map(async (x) => {
-                    if (isProfileIdentifier(x.profile)) {
-                        const res = first(await Services.Identity.queryProfilesInformation([x.profile]))
-                        if (res?.linkedPersona !== undefined && res?.linkedPersona !== currentPersona?.identifier)
-                            return {
-                                persona: res.linkedPersona,
-                                profile: x.profile,
-                                avatar: res.avatar,
-                            }
-                        return
-                    } else {
-                        if (x.profile !== currentPersona?.identifier) return { persona: x.profile }
-                        return
-                    }
-                }),
-            )
-        ).filter(emptyFilter)
-        const allSettled = await Promise.allSettled(
-            friends.map((item) => {
-                const id = item.persona.publicKeyAsHex
-                return NextIDProof.queryProfilesByPublicKey(id, 2)
-            }),
-        )
-        const profiles: FriendsInformation[] = allSettled.map((item, index) => {
-            if (item.status === 'rejected') {
-                if (friends[index].profile) {
-                    return {
-                        profiles: [
-                            {
-                                platform: NextIDPlatform.Twitter,
-                                identity: friends[index].profile!.userId,
-                                is_valid: true,
-                                last_checked_at: '',
-                                name: friends[index].profile!.userId,
-                                created_at: '',
-                            },
-                        ],
-                        ...friends[index],
-                        id: friends[index].persona.publicKeyAsHex,
-                    }
+    return useInfiniteQuery(
+        ['friends', currentPersona],
+        async ({ pageParam }) => {
+            if (!cachedValues.current.length && currentPersona) {
+                const values = await Services.Identity.queryRelationPaged(
+                    currentPersona?.identifier,
+                    {
+                        network: 'all',
+                        pageOffset: 0,
+                    },
+                    3000,
+                )
+                if (values.length === 0) return EMPTY_LIST
+                cachedValues.current = values
+            }
+            const friends: Friend[] = []
+            const startIndex = pageParam ? 10 * Number(pageParam) : 0
+            for (let i = startIndex; i < cachedValues.current.length; i += 1) {
```

Comment:
```ts
const friends = cachedValues.map(() => {
})
```

Reply:
Need to break the loop when there are 10 valid results